compose: Fix numeric typing regression related to global time widget.

YashRE42 opened the pull request:
commit message:
```
In commit 1d54b383bd471af0a45a96b2dcf8d62aec60f516 we introduced some
changes to add better support for keyboard navigation with the global
time widget. Unfortunately, as a result of the fact that
get_keydown_hotkey returns undefined for numeric keys, we caused a
regression that prevented users from typing into the time picker.

Hence, this commit fixes the above bug by checking if the key pressed
is a numeric key.
```
<!-- What's this PR for?  (Just a link to an issue is fine.) -->
https://chat.zulip.org/#narrow/stream/9-issues/topic/typing.20into.20time.20picker

**Testing plan:** <!-- How have you tested? -->
Only manual testing...

**GIFs or screenshots:** <!-- If a UI change.  See:
  https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
  -->
<details>
<summary>
gif
</summary>

![](https://user-images.githubusercontent.com/33805964/146146310-76a5c5bc-dcb6-4afe-9a58-d5695101e1a0.gif)
</details>

<!-- Also be sure to make clear, coherent commits:
  https://zulip.readthedocs.io/en/latest/contributing/version-control.html
  -->

timabbott commented on static/js/composebox_typeahead.js:
```diff
@@ -788,26 +792,28 @@ export function show_flatpickr(element, callback, default_timestamp, options = {
-                const remain = (i + (event.shiftKey ? -1 : 1)) % n;
-                const target = elems[Math.floor(remain >= 0 ? remain : remain + n)];
-                event.preventDefault();
+            if (!is_numeric_key(event.key)) {
```
I'd do an early return here, like you do in the other place.

Also, this doesn't fully fix the bug -- you can't use `backspace` when editing the number inputs.

YashRE42 commented:
Updated to handle backspace and delete.

```
compose: Fix numeric typing regression related to global time widget.

In commit 1d54b383bd471af0a45a96b2dcf8d62aec60f516 we introduced some
changes to add better support for keyboard navigation with the global
time widget. Unfortunately, as a result of the fact that
get_keydown_hotkey returns undefined for numeric keys, we caused a
regression that prevented users from typing into the time picker.
Additionally, we also lost support for the backspace and delete keys.

Hence, this commit fixes the above bug by early returning in two
places if the key pressed is backspace, or delete or a numeric key.
```
<details>
<summary>
gif
</summary>

![](https://user-images.githubusercontent.com/33805964/148035276-856b250e-14f3-48ec-9f6f-7bfbf1eae112.gif)
</details>

timabbott commented:
Merged, thanks @YashRE42!

One note -- I think it's generally more readable to avoid ever using `//` comments on a line with code.  Instead, always put it on the line before the code you're explaining.  We may move to enforcing that convention as a linter rule at some point.

andersk commented on static/js/composebox_typeahead.js:
```diff
@@ -788,8 +792,18 @@ export function show_flatpickr(element, callback, default_timestamp, options = {
+
             const hotkey = get_keydown_hotkey(event);
 
+            if (hotkey === "backspace" || hotkey === "delete") {
```
Was this supposed to be `hotkey.name`?

YashRE42 replied:
yes! I thought I'd changed that, thanks for pointing out.

YashRE42 replied:
#20688
